File: src/main/java/com/SaasRRHH/main/services/PlanillaService.java
package com.SaasRRHH.main.services;

import com.SaasRRHH.main.model.Planilla;

import java.util.List;
import java.util.Optional;

public interface PlanillaService {
    List<Planilla> listar();

    Optional<Planilla> buscarPorId(Long id);

    Planilla guardar (Planilla planilla);

    Planilla actualizar(Long id, Planilla planilla);

    void eliminar(Long id);

}

File: frontend/src/app/core/services/planilla.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Planilla, PlanillaRequest } from '../models/planilla.model';

@Injectable({ providedIn: 'root' })
export class PlanillaService {
  private readonly baseUrl = `${environment.apiUrl}/api/planillas`;

  constructor(private http: HttpClient) {}

  listar(): Observable<Planilla[]> {
    return this.http.get<Planilla[]>(this.baseUrl);
  }

  buscarPorId(id: number): Observable<Planilla> {
    return this.http.get<Planilla>(`${this.baseUrl}/${id}`);
  }

  guardar(planilla: PlanillaRequest): Observable<Planilla> {
    return this.http.post<Planilla>(this.baseUrl, planilla);
  }

  actualizar(id: number, planilla: Partial<PlanillaRequest>): Observable<Planilla> {
    return this.http.put<Planilla>(`${this.baseUrl}/${id}`, planilla);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
